### PR TITLE
feat(documentation): add version-switcher to documentationv7

### PR DESCRIPTION
**Pull request by b1aserlu**
end of day commit

### CODE REVIEW

File: packages/documentation/.storybook/version-switcher/VersionSwitcher.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        setLoading(false);
+      } catch (error) {
+        console.log(`failed to fetch versions file. Errormessage: ${error}`);
+      }
+    }
+    fetchVersions();
+  }, []);
+
+  if (loading) {
+    return <div>Loadinf...</div>;
```
Comment (gfellerph) [inline packages/documentation/.storybook/version-switcher/VersionSwitcher.tsx:46]:
```suggestion
    return <div>Loading...</div>;
```

File: packages/documentation/.storybook/version-switcher/register.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+
+import { addons, types } from '@storybook/addons';
+import VersionSwitcher from './VersionSwitcher';
+
+const ADDON_ID = 'myAddonId';
```
Comment (gfellerph) [inline packages/documentation/.storybook/version-switcher/register.tsx:6]:
```suggestion
const ADDON_ID = 'postVersionSwitcher';
```

File: packages/documentation/.storybook/version-switcher/register.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+addons.register(ADDON_ID, () => {
+  addons.add(ADDON_ID, {
+    title: 'My button title',
```
Comment (gfellerph) [inline packages/documentation/.storybook/version-switcher/register.tsx:10]:
```suggestion
    title: 'Switch to another version',
```

File: packages/documentation/.storybook/version-switcher/VersionSwitcher.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  }
+
+  return (
+    <div>
+      <IconButton
+        title="Versions"
+        placeholder="Versions"
+        onClick={(e: React.MouseEvent<HTMLButtonElement>) => handleClick(e)}
+      >
+        Versions
```
Comment (gfellerph) [inline packages/documentation/.storybook/version-switcher/VersionSwitcher.tsx:56]: This should show the current version as "v[major].[minor]", e.g. "v6.5"

File: packages/documentation/.storybook/styles/version-switcher.scss [hunk 5]
```diff
@@ -0,0 +1,39 @@
+.version-switcher-dropdown {
+  display: flex;
+  flex-flow: column nowrap;
+  background-color: var(--post-light);
+
+  .version-switcher-dropdown-item {
+    border-top: 1px solid var(--post-black);
+    max-width: 100%;
+    padding: 0.5rem 0.75rem;
+    text-decoration: none;
```
Comment (gfellerph) [inline packages/documentation/.storybook/styles/version-switcher.scss:10]:
```suggestion
    text-decoration: none;
    color: inherit;
```

File: packages/documentation/.storybook/styles/version-switcher.scss [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+    &:hover {
+      background-color: var(--post-gray-10);
+    }
+
+    .icons {
+      display: flex;
+      align-items: center;
+      flex-wrap: wrap;
+      gap: 0.75rem;
```
Comment (gfellerph) [inline packages/documentation/.storybook/styles/version-switcher.scss:24]:
```suggestion
      gap: 0.75rem;
      font-size: 0.8em;
```

File: packages/documentation/.storybook/styles/version-switcher.scss [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+      span {
+        display: flex;
+        align-items: center;
+        gap: 0.25rem;
+
+        img {
+          max-width: 1.5rem;
+          min-width: 1.5rem;
+          width: 1.5rem;
```
Comment (gfellerph) [inline packages/documentation/.storybook/styles/version-switcher.scss:34]:
```suggestion
          max-width: 1.5em;
          min-width: 1.5em;
          width: 1.5em;
```

**Review by gfellerph — CHANGES_REQUESTED** [review]

File: packages/demo/src/assets/versions.json [hunk 8]
```diff
@@ -1,14 +1,14 @@
 [
   {
-    "title": "New docs (beta)",
+    "title": "Version 6 — Current docs",
     "description": "Developer documentation for the Swiss Post Design System.",
     "url": "https://next.design-system.post.ch"
   },
```
[the remark this answers is not in the document]
Comment (gfellerph) [reply packages/demo/src/assets/versions.json:6]: I think we can remove this entry. When we switch Storybook to https://design-system.post.ch, there will be no next version to publish.

File: packages/documentation/src/utils/version.ts [hunk 9]
```diff
@@ -29,7 +29,7 @@ const versionFilterMap: any = {
 };
 
 export function getVersion(version: string, filter: string = '') {
-  const cleanVersion = DEPENCENCIES[version].replace(/^[^\d]+/, '');
+  const cleanVersion = version.replace(/^[^\d]+/, '');
```
Comment (gfellerph) [inline packages/documentation/src/utils/version.ts:32]: Not sure, but I think this breaks the version extraction:
image

**Review by gfellerph — CHANGES_REQUESTED** [review]

File: packages/documentation/src/utils/version.ts (hunk 9 above)
Comment (imagoiq) [reply packages/documentation/src/utils/version.ts:32]: Good catch!

**Review by gfellerph — APPROVED** [review]

**Pull request merged**